lib/bundler.rb

```diff
@@ -143,8 +145,41 @@ def ruby_scope
       "#{Bundler.rubygems.ruby_engine}/#{Bundler.rubygems.config_map[:ruby_version]}"
     end
 
+    def user_home
+      @user_home ||= begin
+        home = Bundler.rubygems.user_home
+        warning = "Your home directory is not set properly:"
+        if home.nil?
+          warning += "\n * It is not set at all"
+        elsif !File.directory?(home)
+          warning += "\n * `#{home}` is not a directory"
+        elsif !File.writable?(home)
+          warning += "\n * `#{home}` is not writable"
+        else
+          return Pathname.new(home)
```

Comment:
This will interrupt `begin` block, return `Pathname.new(home)` and will leave `@user_home.nil?`